cmake/ci.cmake

```diff
@@ -577,6 +584,13 @@ add_custom_target(ci_test_amalgamation
     COMMAND ${ASTYLE_TOOL} ${ASTYLE_FLAGS} ${INDENT_FILES}
     COMMAND cd ${PROJECT_SOURCE_DIR} && for FILE in `find . -name '*.orig'`\; do false \; done
 
+    COMMAND CXX=${GCC_TOOL} CXXFLAGS="${GCC_CXXFLAGS}" ${CMAKE_COMMAND}
```

Comment:
Why did you add this to `ci_test_amalgamation`?

Reply:
It seemed related. Amalgamation should be the only reason why the single-header could possibly behave differently than the multi-header version.

Reply:
But the test only aims at the structure. The goal is not to detect different behavior, but is to make sure changes are applied to both the `include` and the `single_include` folder.

Reply:
I guess I just wanted to avoid adding another job to the workflow.

Look forward to an issue (and PR) on `json-ci` soon. Preview:
> The `json-ci` image is about 14GB in size. The time spent initializing containers is 170 seconds (average over ~350 jobs) or almost 3 minutes. That amounts to a ridiculous 2.5 hours per workflow run.

Reply:
Oh I so agree! We need to cut off the easy targets from the harder ones. I guess 2-3 Debian images could be sufficient to cover 90% of the compilers. Then we'd only need 1 image for the bleeding edge stuff and the specific tools.

Reply:
Do you want a test for the single-header at all?

Reply:
Good point. So far, we tested it by chance somewhere, because we did not switch on the multiple-header version in all of the tests. Now that we have the multiple-header version as default, a dedicated test for the amalgamated version should be in order.